### PR TITLE
chore: add mixed protocol service e2e test

### LINKED ISSUE #873: Add e2e tests for mixed protocols services
Refer https://kubernetes.io/docs/concepts/services-networking/service/#load-balancers-with-mixed-protocol-types, e2e tests for mixed protocols services should be added.

> FEATURE STATE: Kubernetes v1.20 [alpha]
>
> By default, for LoadBalancer type of Services, when there is more than one port defined, all ports must have the same protocol, and the protocol must be one which is supported by the cloud provider.
>
> If the feature gate MixedProtocolLBService is enabled for the kube-apiserver it is allowed to use different protocols when there is more than one port defined.

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

chore: add mixed protocol service e2e test

#### Which issue(s) this PR fixes:

Fixes #873 

#### Does this PR introduce a user-facing change?

```release-note
chore: add mixed protocol service e2e test
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/network/ensureloadbalancer.go [hunk 1]
```diff
@@ -88,6 +88,47 @@ var _ = Describe("Ensure LoadBalancer", func() {
 		tc = nil
 	})
 
+	FIt("should support mixed protocol services", func() {
```
Comment (feiskyer) [inline tests/e2e/network/ensureloadbalancer.go:91]: do we need to enable feature gates on capz and aks-engine cluster configs?

Comment (nilo19) [reply tests/e2e/network/ensureloadbalancer.go:91]: aks-engine enabled in https://github.com/kubernetes-sigs/cloud-provider-azure/pull/898
capz pending for https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1847

**Review by feiskyer — APPROVED** [review]
/lgtm

**Comment by nilo19** [comment]
capz test pending for: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1851

**Comment by CecileRobertMichon** [comment]
/retest

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1851 just merged

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**